tests/components/mqtt/test_init.py

```diff
@@ -1501,22 +1495,26 @@ async def test_replaying_payload_same_topic(
     Retained messages must only be replayed for new subscriptions, except
     when the MQTT client is reconnecting.
     """
-    await mqtt_mock_entry()
-
+    mqtt_client_mock = setup_with_birth_msg_client_mock
     calls_a: list[ReceiveMessage] = []
     calls_b: list[ReceiveMessage] = []
 
+    @callback
     def _callback_a(msg: ReceiveMessage) -> None:
         calls_a.append(msg)
 
+    @callback
     def _callback_b(msg: ReceiveMessage) -> None:
         calls_b.append(msg)
 
+    mqtt_client_mock.reset_mock()
     await mqtt.async_subscribe(hass, "test/state", _callback_a)
+    await hass.async_block_till_done()
     async_fire_mqtt_message(
         hass, "test/state", "online", qos=0, retain=True
     )  # Simulate a (retained) message played back
-    await hass.async_block_till_done()
+    async_fire_time_changed(hass, utcnow() + timedelta(seconds=3))
+    await asyncio.sleep(0.2)
```

Comment:
Can you get rid of the sleeps here? Or rework so it's sleep(0) to yield to the loop instead ?